src/subdomains/generic/kyc/services/kyc.service.ts

```diff
@@ -656,6 +669,45 @@ export class KycService {
     return userData;
   }
 
+  async completeSumsubIdent(result: SumsubResult, userData: UserData, nationality?: Country): Promise<UserData> {
+    const identificationType = getIdentificationType(result.webhook.type);
+    if (
+      result.data.info?.idDocs?.[0]?.dob &&
+      result.data.info?.idDocs?.[0]?.country &&
+      identificationType &&
+      result.data.info?.idDocs?.[0]?.number &&
+      result.data.info?.idDocs?.[0]?.idDocType &&
+      nationality
+    ) {
+      const identDocumentId = `${userData.organizationName?.split(' ')?.join('') ?? ''}${
+        result.data.info.idDocs[0].number
+      }`;
+      const existing = await this.userDataService.getDifferentUserWithSameIdentDoc(userData.id, identDocumentId);
+
+      if (existing) {
+        await this.accountMergeService.sendMergeRequest(existing, userData);
+
+        return userData;
+      } else if (nationality) {
+        return this.userDataService.updateUserDataInternal(userData, {
+          kycLevel: KycLevel.LEVEL_30,
+          birthday: new Date(result.data.info.idDocs[0].dob),
+          verifiedCountry: !userData.verifiedCountry ? userData.country : undefined,
+          identificationType,
+          bankTransactionVerification:
+            identificationType === KycIdentificationType.VIDEO_ID ? CheckStatus.UNNECESSARY : undefined,
+          identDocumentType: result.data.info.idDocs[0].idDocType === IdDocType.ID_CARD ? 'IDCARD' : 'PASSPORT',
+          identDocumentId,
+          nationality,
+        });
+      }
```

Comment:
This looks the same than for Intrum. Can we combine it? Maybe create a method to get all required fields?

Reply:
Aha, yes, good point